src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,85 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int index = hash(key);
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value, null);
+        } else {
+            Node<K, V> currentNode = table[index];
+            while (currentNode != null) {
+                if (Objects.equals(currentNode.key, key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                if (currentNode.next == null) {
+                    break;
+                }
+                currentNode = currentNode.next;
+            }
+            currentNode.next = new Node<>(key, value, null);
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = hash(key);
+        Node<K, V> currentNode = table[index];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        if (size > table.length * LOAD_FACTOR) {
+            Node<K, V>[] oldTable = table;
+            table = new Node[oldTable.length << 1];
+            size = 0;
+            for (Node<K, V> element : oldTable) {
+                while (element != null) {
+                    put(element.key, element.value);
+                    element = element.next;
+                }
+            }
+        }
+    }
+
+    private int hash(Object key) {
```

Comment:
is not a valid name, because from the method you return the basket index, not the hash